sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2264,4 +2264,18 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     checkAnswer(df, Row(0, 10) :: Nil)
     assert(df.queryExecution.executedPlan.isInstanceOf[WholeStageCodegenExec])
   }
+
+  test("SPARK-23531: explain should show attributes' type") {
```

Comment:
Hi, @mgaido91 .
For the test case, could you
- Add a test coverage for `explain(true)`, too?
- Try to add `explain.sql` into `SQLQueryTestSuite`?

cc @gatorsmile

Reply:
thanks for the review @dongjoon-hyun, I added it too